### PR TITLE
Web api minimal hosting model doc migration

### PR DESCRIPTION
**Pull request by LayersOfAbstraction**
Hi as requested in [the #79 issue outside of this repo](https://github.com/auth0-samples/auth0-aspnetcore-mvc-samples/issues/79) I had edited the files for the Web API quickstart you mentioned. Here are the doc enhancements. 

FYI:
- All I did was swap in the markdown files for the Startup code for how to use the minimal hosting model logic in Program.cs. 
- I cannot submit this to Wordy unless I am reimbursed for submitting this to one of their editors. I imagine that's not even an issue. 

Hope it all looks good! We'll see.

### CODE REVIEW

File: articles/quickstart/backend/aspnet-core-webapi/01-authorization.md [hunk 1]
```diff
@@ -140,24 +127,18 @@ public class HasScopeHandler : AuthorizationHandler<HasScopeRequirement>
 }
 ```
 
-In your Startup's `ConfigureServices` method, add a call to the `AddAuthorization` method. To add policies for the scopes, call `AddPolicy` for each scope. Also ensure that you register the `HasScopeHandler` as a singleton:
+Under your Program's `var builder = WebApplication.CreateBuilder(args);` method, call the `AddAuthorization` method. To add policies for the scopes, call `AddPolicy` for each scope. Also ensure that you register the `HasScopeHandler` as a singleton:
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/01-authorization.md]:
```suggestion
In your Program.cs file, add a call to the `AddAuthorization` method. To add policies for the scopes, call `AddPolicy` for each scope. Also ensure that you register the `HasScopeHandler` as a singleton:
```

File: articles/quickstart/backend/aspnet-core-webapi/01-authorization.md [hunk 2]
```diff
@@ -54,45 +54,32 @@ In your application, register the authentication services:
...
+    options.Audience = builder.Configuration["Auth0:Audience"];
+    options.TokenValidationParameters = new TokenValidationParameters
+    {
+        NameClaimType = ClaimTypes.NameIdentifier
+    };
+});
 ```
 
-To add the authentication and authorization middleware to the middleware pipeline, add a call to the `UseAuthentication` and `UseAuthorization` methods in your Startup's `Configure` method:
+To add the authentication and authorization middleware to the middleware pipeline, call  the `UseAuthentication` and `UseAuthorization` methods under your Program.cs `var app = builder.Build();` method:
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/01-authorization.md]:
```suggestion
To add the authentication and authorization middleware to the middleware pipeline, add a call to the `UseAuthentication` and `UseAuthorization` methods in your Program.cs file:
```

File: articles/quickstart/backend/aspnet-core-webapi/01-authorization.md [hunk 3]
```diff
@@ -54,45 +54,32 @@ In your application, register the authentication services:
...
 
 ```csharp
 // Startup.cs
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/01-authorization.md]:
```suggestion
// Program.cs
```

File: articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md [hunk 4]
```diff
@@ -33,10 +29,18 @@ public void ConfigureServices(IServiceCollection services)
     services.AddSingleton<IAuthorizationHandler, HasScopeHandler>();
 }
 
-public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+builder.Services.AddAuthorization(options =>
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md]: We should remove the code above this one.

File: articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md [hunk 5]
```diff
@@ -33,10 +29,18 @@ public void ConfigureServices(IServiceCollection services)
...
-}
+    options.AddPolicy(
+        "read:messages",
+         policy => policy.Requirements.Add(
+            new HasScopeRequirement("read:messages", domain)
+        )
+    );
+});
+
+services.AddSingleton<IAuthorizationHandler, HasScopeHandler>();
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md]:
```suggestion
builder.Services.AddSingleton<IAuthorizationHandler, HasScopeHandler>();
```

File: articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md [hunk 6]
```diff
@@ -4,20 +4,16 @@ language: csharp
 ---
 
 ```csharp
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md:6]: I cant comment on line 2, but we need to rename this to Program.cs

**Review by frederikprijck — CHANGES_REQUESTED** [review]
Thanks for the PR!

I believe there are still some things that needs to be changed in the text part of it. E.g. on https://auth0.com/docs/quickstart/backend/aspnet-core-webapi/interactive you can see see references to Startup.cs in the text, Can you please update these as well?

File: articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md (hunk 6 above)
Comment (LayersOfAbstraction) [reply articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md:6]: Gotya. Eliminated syntax errors and renamed frontmatter name from Startup.cs to Program.cs.

File: articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md (hunk 4 above)
Comment (LayersOfAbstraction) [reply articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md]: Done. Had a duplicate code block. Removed that.

File: articles/quickstart/backend/aspnet-core-webapi/01-authorization.md [hunk 7]
```diff
@@ -54,45 +54,32 @@ In your application, register the authentication services:
 2. Make a call to the `AddJwtBearer` method to register the JWT Bearer authentication scheme. Configure your Auth0 domain as the authority, and your Auth0 API identifier as the audience. In some cases the access token will not have a `sub` claim which will lead to `User.Identity.Name` being `null`. If you want to map a different claim to `User.Identity.Name` then add it to `options.TokenValidationParameters` within the `AddAuthentication()` call.
 
 ```csharp
-// Startup.cs
+// Program.cs
 
-public void ConfigureServices(IServiceCollection services)
+var domain = $"https://{builder.Configuration["Auth0:Domain"]}/";
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/01-authorization.md:59]: Can we add what builder is here ? So let's add `var builder = WebApplication.CreateBuilder(args);`

File: articles/quickstart/backend/aspnet-core-webapi/01-authorization.md [hunk 8]
```diff
@@ -54,45 +54,32 @@ In your application, register the authentication services:
...
 
-To add the authentication and authorization middleware to the middleware pipeline, add a call to the `UseAuthentication` and `UseAuthorization` methods in your Startup's `Configure` method:
+To add the authentication and authorization middleware to the middleware pipeline, add a call to the `UseAuthentication` and `UseAuthorization` methods in your Program.cs file:
 
 ```csharp
-// Startup.cs
+// Program.cs
 
-public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+app.UseAuthentication();
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/01-authorization.md:78]: Can we add what `app` is here ?

File: articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md [hunk 9]
```diff
@@ -1,25 +1,21 @@
 ---
-name: Startup.cs
+name: Program.cs
 language: csharp
 ---
 
 ```csharp
-public void ConfigureServices(IServiceCollection services)
+builder.Services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md:8]: Can we add what builder is here ? So let's add `var builder = WebApplication.CreateBuilder(args);`

File: articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md [hunk 10]
```diff
@@ -30,13 +26,8 @@ public void ConfigureServices(IServiceCollection services)
...
-}
+    builder.Services.AddSingleton<IAuthorizationHandler, HasScopeHandler>();
 
-public void Configure(IApplicationBuilder app, IHostingEnvironment env)
-{
-  // Some code omitted for brevity...
-  app.UseAuthentication();
-  app.UseAuthorization();
-}
+app.UseAuthentication();
```
Comment (frederikprijck) [inline articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md:33]: Can we add what app is here ?

**Review by frederikprijck — CHANGES_REQUESTED** [review]

File: articles/quickstart/backend/aspnet-core-webapi/01-authorization.md (hunk 7 above)
Comment (LayersOfAbstraction) [reply articles/quickstart/backend/aspnet-core-webapi/01-authorization.md:59]: Done as per [501677e](https://github.com/auth0/docs/pull/10231/commits/501677e349cf36e85c17f6bd1c4a5c96d19ea441).

File: articles/quickstart/backend/aspnet-core-webapi/01-authorization.md (hunk 8 above)
Comment (LayersOfAbstraction) [reply articles/quickstart/backend/aspnet-core-webapi/01-authorization.md:78]: You mean like add that `var app = builder.build();` statement? 
I did so as per [0a5e178](https://github.com/auth0/docs/pull/10231/commits/0a5e178ac640d1b904f06ef94a61648900e677d3). Let me know if I missed anything.

File: articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md (hunk 9 above)
Comment (LayersOfAbstraction) [reply articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md:8]: Done as per [45e34bb](https://github.com/auth0/docs/pull/10231/commits/45e34bbc83210bdde43c5cbaf4c91899710e3a42).

File: articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md (hunk 10 above)
Comment (LayersOfAbstraction) [reply articles/quickstart/backend/aspnet-core-webapi/files/configure-middleware.md:33]: Gotya! All done. Please see https://github.com/auth0/docs/pull/10231/commits/71b753076a1e4d54f2771f9b0e4cda22030575f4. I will let you resolve all conversations yourself to make sure I didn't miss anything. Thanks.

**Review by frederikprijck — CHANGES_REQUESTED** [review]
Thanks, looks very good! It does look like we missed updating the `interactive.md` file here, could you update it as well?

**Comment by LayersOfAbstraction** [comment]
Whoops, that one file you mentioned before. Done it now as per  https://github.com/auth0/docs/pull/10231/commits/c596f67b1b2a61c118517a75ff760de2830ae926.

**Review by frederikprijck — APPROVED** [review]

**Pull request merged**